app/scripts/modules/core/widgets/uiSelect.component.ts

```diff
@@ -0,0 +1,55 @@
+import {Directive, ElementRef, Injector, Input} from '@angular/core';
+import {UpgradeComponent} from '@angular/upgrade/static';
+import {module, IComponentOptions} from 'angular';
+
+class UiSelectComponent implements IComponentOptions {
+
+  public bindings: any = {
+    items: '<',
+    model: '<',
+    modelProperty: '@',
+    placeholder: '@',
+    renderProperty: '@',
+    selectProperty: '@'
+  };
+  public template = `
+    <ui-select ng-model="$ctrl.model[$ctrl.modelProperty]" class="form-control input-sm" required>
+      <ui-select-match placeholder="{{$ctrl.placeholder}}">{{$select.selected[$ctrl.renderProperty]}}</ui-select-match>
+      <ui-select-choices repeat="item[$ctrl.selectProperty] as item in $ctrl.items | filter: $select.search">
+        {{item[$ctrl.renderProperty]}}
```

Comment:
A lot of (but not all) our ui-select widgets include `| highlight: $select.search`. It would be nice to make that available.

How do you see addressing formatted select options, where the rendered value is not just a plain property? For example, the `deploymentStrategySelector` directive uses:
```html
<ui-select-choices repeat="strategy.key as strategy in deploymentStrategies | filter: $select.search">
  <strong ng-bind-html="strategy.label | highlight: $select.search"></strong>
  <div ng-bind-html="strategy.description"></div>
</ui-select-choices>
```